src/benchmarks/micro/libraries/System.IO/BinaryWriterExtendedTests.cs

```diff
@@ -0,0 +1,41 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using BenchmarkDotNet.Attributes;
+using MicroBenchmarks;
+
+namespace System.IO.Tests
+{
+    [BenchmarkCategory(Categories.Libraries)]
+    public class BinaryWriterExtendedTests
+    {
+        private string _input;
+        private char[] _inputAsChars;
+        private BinaryWriter _bw;
+
+        [Params(4, 16, 512, 10_000, 100_000, 500_000, 2_000_000)]
```

Comment:
can we reduce the number of test cases to the number of different code paths? https://github.com/dotnet/performance/blob/master/docs/microbenchmark-design-guidelines.md#Code-Paths

Reply:
Sure, there should be 3 different code paths. How about 32 chars, 8k chars, and 2M chars?

Reply:
>How about 32 chars, 8k chars, and 2M chars?

👍